backends/dnf/pk-backend-dnf.c

```diff
@@ -150,10 +152,16 @@ pk_backend_setup_dnf_context (DnfContext *context, GKeyFile *conf, const gchar *
 	dnf_context_set_cache_dir (context, cache_dir);
 	solv_dir = g_build_filename (destdir, "/var/cache/PackageKit", release_ver, "hawkey", NULL);
 	dnf_context_set_solv_dir (context, solv_dir);
-	repo_dir = g_build_filename (destdir, "/etc/yum.repos.d", NULL);
-	dnf_context_set_repo_dir (context, repo_dir);
 	lock_dir = g_build_filename (destdir, "/var/run", NULL);
 	dnf_context_set_lock_dir (context, lock_dir);
+	repo_dirs = dnf_context_get_repos_dir (context);
+	for (diriter = 0; repo_dirs[diriter] != NULL; diriter++)
```

Comment:
`for (guint i = 0...` please

Reply:
I prefer that myself, but I saw other code in here doing otherwise, so...

Reply:
Fixed.